ToDoApp/ToDoApp/ToDoListView.swift

```diff
@@ -21,7 +21,12 @@ struct ToDoListView: View {
                 case .idle:
                     EmptyView()
                 case .failed(let error):
-                    Text(error.localizedDescription)
+                    VStack(spacing: 20) {
+                        Text(error.localizedDescription)
+                        Button("Retry") {
+                            viewModel.onAppear()
```

Comment:
It's not the best solution to invoke the on Appear method of the view model. Even though it seems they are doing the same thing. If we would need to extend the on Appear implementation with additional functionality which may not be wanted for the retry button tap handling. And it would not be obvious that method on Appear is used by something else except by the on Appear modifier

Reply:
so should I have another method like retryButtonTapped(), in which it will continuously call fetchToDoTasks() until the state is loaded?